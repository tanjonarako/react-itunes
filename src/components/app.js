import React, { Component } from 'react';
import SearchBar from './search_bar';
import SearchResultList from './search_result_list';
import '../stylesheet/app.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <SearchBar/>
        <SearchResultList/>
      </div>
    );
  }
}

export default App;
