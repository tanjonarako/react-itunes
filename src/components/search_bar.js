import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const searchAction = (searchTerm) => ({
  type: "updateList",
  payload: {
    request: {
      url: 'https://itunes.apple.com/search?term=' + searchTerm
    }
  }
});

const mapDispatchToProps = (dispatch) => ({
  doSearch: (searchTerm) => dispatch(searchAction(searchTerm)),
});

export class SearchBar extends Component {
  state = {
    value: ''
  }

  static propTypes = {
    doSearch: PropTypes.func
  }

  handleSubmit = (e) => { 
    e && e.preventDefault();

    const inputValue = this.state.value;

    this.props.doSearch(inputValue);
    this.setState({value: ''});
  }

  handleChange = (e) => {
    this.setState({value: e.target.value});
  }

  render() {
    return (
      <header className="justify-content-center align-items-center search-bar">
        <div className="navbar bg-nav row-flex">
          <p className="navbar-brand m-0 py-0 pr-2" href="#">
            <img src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg" width="40" height="40" className="d-inline-block align-center" alt="itunes" />
            iTunes
          </p>
          <form className="search-form my-2 my-lg-0 col-flex" onSubmit={this.handleSubmit} >
            <input className="search-form__input bg-dark" type="search" placeholder="Search" value={this.state.value} onChange={this.handleChange} />
            <a className="search-form__button m-0 p-0" href="#" role="button" type="submit" onClick={this.handleSubmit} ><img src="https://icongr.am/clarity/search.svg?color=FFFFFF" width="20" alt="search icon"/></a>
          </form>
        </div>
      </header>
    );
  }
}

export default connect(
  state => ({}),
  mapDispatchToProps
)(SearchBar)
