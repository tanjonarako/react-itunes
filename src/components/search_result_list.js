import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchResultListItem from './search_result_list_item';


export class SearchResultList extends Component {
  static propTypes = {
    list: PropTypes.array
  }

  render() {
    const list = this.props.list;
    return (
      <div className="search-result-list">
        <ul className="list-unstyled">
          {list.map((item) => (
            <SearchResultListItem key={item.trackId} item={item}/>
          ))}
        </ul>
      </div>
    );
  }
}

SearchResultList.defaultProps = { list: []}

export default connect(
  state => ({
    list: state.list
  })
)(SearchResultList)
