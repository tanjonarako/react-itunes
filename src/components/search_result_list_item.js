import React from 'react';

const SearchResultListItem = (props) => (
  <li className="search-result-list__song row-flex p-3">
    <img src={props.item.artworkUrl60} width="60" height="60" alt="song logo"/>
    <div className="search-result-list__song-details pl-3 col-flex">
      <h2 className="search-result-list__song-name m-0">{props.item.trackName}</h2>
      <span className="search-result-list__song-artist m-0">{props.item.artistName}</span>
    </div>
  </li>
);

export default SearchResultListItem;
