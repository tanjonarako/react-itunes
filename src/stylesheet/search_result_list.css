.search-result-list {
  margin:0 auto;
  position:absolute;
  overflow: auto;
  top: 62px;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: left;
}
.search-result-list__song {
  border-bottom: 1px solid #f3f3f3;
}

.search-result-list__song-artist {
  color: #888;
}

.search-result-list__song-name {
  font-size: 18px;
}
