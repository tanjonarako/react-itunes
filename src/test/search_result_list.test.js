import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SearchResultList } from '../components/search_result_list';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchResultList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SearchResultList/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list', () => {
    const list = [{trackId: 1, name: "Shape of you"}, {trackId: 2, name: "Dive"}];
    const wrapper = shallow(<SearchResultList list={ list }/>);
    expect(wrapper.find('SearchResultListItem').length).toEqual(2);
  }); 
});
