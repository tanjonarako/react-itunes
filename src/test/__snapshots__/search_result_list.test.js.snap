// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultList
    list={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list-unstyled"
>
        
</ul>,
      "className": "search-result-list",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "list-unstyled",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list-unstyled"
>
          
</ul>,
        "className": "search-result-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "list-unstyled",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
